manager/src/main/java/io/gingersnapproject/search/QueryHandler.java

```diff
@@ -0,0 +1,25 @@
+package io.gingersnapproject.search;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import io.gingersnapproject.Caches;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@Singleton
+public class QueryHandler {
+   @Inject
+   Caches caches;
+
+   @Inject
+   SearchBackend searchBackend;
+
+   public Multi<String> query(String query) {
```

Comment:
Maybe we should expose the hitCount too. See https://infinispan.zulipchat.com/#narrow/stream/313207-memory-service/topic/Cache-Manager/near/316208754

Reply:
I would like to add that in a different PR, so this can be integrated. A bunch of other PRs hinge on this.

Reply:
Sure. Feel free to rebase and merge from me.